internal/chain/engine.go

```diff
@@ -360,19 +354,13 @@ func setLastEpochInCommittee(header *block.Header, state *state.DB) error {
 		return errors.New(msg)
 	}
 	for _, addr := range newShardState.StakedValidators().Addrs {
-		wrapper, err := state.ValidatorWrapper(addr)
+		wrapper, err := state.ValidatorWrapper(addr, false)
 		if err != nil {
 			return errors.New(
 				"[Finalize] failed to get validator from state to finalize",
 			)
 		}
 		wrapper.LastEpochInCommittee = newShardState.Epoch
-		if err := state.UpdateValidatorWrapper(
```

Comment:
I guess all these updates were redundant ?

Reply:
unless the change may fail and need to revert (in staking txn case). Other cases can directly update on the live validatorWrapper object.